feat: add certifications and ratings

flagrede opened the pull request:
## Description

Closes: regen-network/regen-registry#1710
Sanity PR: https://github.com/regen-network/regen-sanity/pull/43

Add:
- RoundLogo (atom)
- RoundLogoItemsList (molecules)
- wire Sanity requests to get certification and rating icons
- edit metadata to handle certification and rating
- wire new fields in `ProjectTopSection`


---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1. Registry: https://deploy-preview-1942--regen-registry.netlify.app/project/C45-007
2. Storybook: https://deploy-preview-1942--regen-storybook.netlify.app/?path=/story/molecules-roundlogoitemslist--default

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

flagrede commented:
@erikalogie you can test this task, I've put links in the description.
Note that I'm not able to edit credit class metadata so I don't have an example with certification but can check the storybook link to get the display with 2 items.

erikalogie commented:
The ratings and certs shouldn't be inside of the grey box FYI

Can we allow for a link to the specific rating or cert?

blushi commented on web-registry/src/lib/db/types/json-ld/certification.ts:
```diff
@@ -0,0 +1,4 @@
+export interface Certification {
```
this structure should be documented and approved in the [standards repo](https://github.com/regen-network/regen-registry-standards/) first, do you feel confortable doing that or should @wgwz or myself handle that?

flagrede replied:
I don't mind doing it but I might need some guidance since this would be the first time.

blushi commented on web-registry/src/lib/db/types/json-ld/toucan-project-metadata.ts:
```diff
@@ -15,6 +17,8 @@ export interface ToucanProjectMetadataLD
   'regen:toucanProjectTokenId': number;
   'regen:toucanURI': string;
+  'regen:rating': BeZeroRating;
+  'regen:certification': Certification;
```
I believe the certification is meant to be stored at the credit class level, see issue description https://github.com/regen-network/regen-registry/issues/1710

flagrede replied:
You're right, will fix it.

flagrede replied:
I moved `regen:certification` to the credit class metadata: [bca7d5c](https://github.com/regen-network/regen-web/pull/1942/commits/bca7d5c74174c981ac80b76fc66611ef44382ad8)
I moved it to `CreditClassMetadataLD` as an optional field. Should we be more explicit and put it as a mandatory field in `CFCCreditClassMetadataLD` and `ToucanCreditClassMetadataLD` (which do not exist currently)?

flagrede commented on web-registry/src/lib/db/types/json-ld/toucan-project-metadata.ts:
```diff
@@ -15,6 +17,8 @@ export interface ToucanProjectMetadataLD
   'regen:approvedMethodologies': ApprovedMethodologies;
   'regen:toucanProjectTokenId': number;
   'regen:toucanURI': string;
+  'regen:rating': BeZeroRating;
```
Is it the same thing for `certification` then?

blushi replied:
I would say so

flagrede commented:
@blushi I've moved both `rating` and `certification` to their base metadata files and also changed their type to list, let me know if that looks good to you: [5484f79](https://github.com/regen-network/regen-web/pull/1942/commits/5484f7985ac1284457a86e377e515126d6d853ab)

flagrede commented:
@erikalogie I've moved the ratings and certification out of the card, handle links (you can test that the first one has one), and also added the case with the fallback icon.

erikalogie commented:
@flagrede is this a generic link or can it be customized per project?

Looks good!

flagrede commented:
@erikalogie it comes from the metadata.

erikalogie commented:
> @erikalogie it comes from the metadata.

Ok cool!

blushi commented on web-registry/src/lib/rdf/rdf.constants.ts:
```diff
@@ -80,6 +80,7 @@ export const COMPACTED_CONTEXT = {
   'schema:itemListElement': { '@container': '@list' },
   'regen:additionalCertifications': { '@container': '@list' },
   'regen:cfcVintageYear': { '@type': 'xsd:gYear' },
+  'regen:rating': { '@container': '@list' },
```
let's add certification as well here

blushi commented on web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.types.ts:
```diff
@@ -48,3 +48,14 @@ export type ProjectTopSectionProps = {
 export type SdgType = SdgByIriQuery['allSdg'][0];
+
+export type ProjectRating = {
+  'schema:name'?: string;
+  'schema:ratingValue'?: string;
+  'schema:url'?: string;
```
why not use the existing `Rating` from rating.ts?

flagrede replied:
Same as certification.

blushi commented on web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.types.ts:
```diff
@@ -48,3 +48,14 @@ export type ProjectTopSectionProps = {
+  'schema:url'?: string;
+};
+
+export type CreditCertification = {
```
also why not using the existing type from certification.ts?

flagrede replied:
I agree, I don't remember why I end creating an other one but it's no longer necessary, will reuse the one from `certification.ts`.

blushi commented:
Logic looks good, just a few nits
